### PR TITLE
Prolong messages to email

### PR DESCRIPTION
**Pull request by milkavladislav**

### CODE REVIEW

File: src/modules/task.service.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { MailerService } from '@nestjs-modules/mailer';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CHAT } from '../utils/auth.consts';
```
Comment (jphinning) [inline src/modules/task.service.ts]: Absolute path

File: src/modules/task.service.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+export class TasksService {
+  constructor(
+    @InjectRepository(ChatRoom)
+    private readonly chatRoomRepository: Repository<ChatRoom>,
+    private readonly mailService: MailerService,
+    private readonly configService: ConfigService,
+  ) {}
+
+  @Cron(CronExpression.EVERY_MINUTE) //EVERY_DAY_AT_NOON
+  async some() {
```
Comment (jphinning) [inline src/modules/task.service.ts]: return type

File: src/modules/task.service.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  dateToSQLDate(date: Date): string {
+    return `"${date.toLocaleDateString().split('.').reverse().join('-')}"`;
+  }
+
+  nextDate(date: Date): Date {
+    return new Date(date.setDate(date.getDate() + 1));
+  }
+
+  sendProlongEmailToUsers(rooms: ChatRoom[]) {
```
Comment (jphinning) [inline src/modules/task.service.ts]: return type

Comment (jphinning) [reply src/modules/task.service.ts]: Can you point out if the functions in this class are public/private as well?

File: src/modules/chat-room/chat-room.service.ts [hunk 4]
```diff
@@ -27,6 +28,8 @@ export class ChatRoomService {
     private readonly usersRepository: Repository<Users>,
   ) {}
 
+  private saltRounds = 5;
```
Comment (lyudad) [inline src/modules/chat-room/chat-room.service.ts]: I think it make sense have it out of class, this value does not belong to specific class. make it constant and make all letters uppercase

File: src/modules/chat-room/chat-room.service.ts [hunk 5]
```diff
@@ -186,6 +189,23 @@ export class ChatRoomService {
...
 
+  async prolongChat(id: number, token: string): Promise<UpdateResult> {
+    const salt = await genSalt(this.saltRounds);
+    const prolongLink = await hash('prolong' + id, salt);
+
+    const result = await this.chatRoomRepository
+      .createQueryBuilder()
+      .update(ChatRoom)
+      .set({
+        deleteDate: new Date(new Date().setMonth(new Date().getMonth() + 6)),
```
Comment (lyudad) [inline src/modules/chat-room/chat-room.service.ts]: what is 6, move this magic number

File: src/modules/contracts/contracts.service.ts [hunk 6]
```diff
@@ -4,21 +4,59 @@ import { CreateContractDto } from 'src/modules/contracts/dto/create-contract.dto
...
 @Injectable()
 export class ContractsService {
   constructor(
     @InjectRepository(Contracts)
     private readonly contractRepo: Repository<Contracts>,
+    @InjectRepository(ChatRoom)
+    private readonly chatRoomRepo: Repository<ChatRoom>,
   ) {}
 
+  private saltRounds = 5;
```
Comment (lyudad) [inline src/modules/contracts/contracts.service.ts]: you create same constant, why this duplication needed, move to one file

File: src/modules/contracts/contracts.service.ts [hunk 7]
```diff
@@ -4,21 +4,59 @@ import { CreateContractDto } from 'src/modules/contracts/dto/create-contract.dto
...
+
   async create(createContractDto: CreateContractDto): Promise<InsertResult> {
-    return await this.contractRepo
+    const result = await this.contractRepo
```
Comment (lyudad) [inline src/modules/contracts/contracts.service.ts:21]: where try/catch?

**Pull request merged**